subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -180,14 +296,23 @@ static void free_segments(void)
 
 		/* Terminate active adv */
 		if (adv->ctx.busy == 0U) {
-			bt_mesh_adv_terminate(adv);
+			err = bt_mesh_adv_terminate(adv);
```

Comment:
```suggestion
			canceled = !bt_mesh_adv_terminate(adv);
```

Reply:
This may change `cancel` back from `true` to `false`.